### PR TITLE
Complete refactoring

### PR DESCRIPTION
**Pull request by corentincarton**
Let's use this branch as a base for the future release. Develop is up to date with feature/prerun, we may push some changes for cutmaps.

### CODE REVIEW

File: integration/check_kge.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    parser.add_argument('target2', default='1.00', help='higher target value')
+    args = parser.parse_args()
+
+    tol = float(args.tol)
+    target1 = float(args.target1)
+    target2 = float(args.target2)
+    print(f'KGE file: {args.kge_file}')
+    print(f'Tolerance is {tol}')
+    print(f'Target 1 is {target1}')
+    print(f'Target 1 is {target2}')
```
Comment (doc78) [inline integration/check_kge.py:32]: Shouldn't it be "Target 2 is..."?

File: integration/check_kge.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    # check KGE
+    check_kge(args.kge_file, target1=target1, target2=target1, tol=2*tol)
```
Comment (doc78) [inline integration/check_kge.py:34]: Shouldn't it be "target1=target1, target2=target2"?

File: liscal/hydro_model.py [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+    prerun_end = cfg.prerun_end.strftime('%d/%m/%Y %H:%M')
+    run_start = cfg.forcing_start.strftime('%d/%m/%Y %H:%M')
+    run_end = cfg.forcing_end.strftime('%d/%m/%Y %H:%M')
+    prerun_file, run_file = lis_template.write_template(run_id, prerun_start, prerun_end, run_start, run_end, cfg.param_ranges, parameters, write_states=True)
+
+    # FIRST LISFLOOD RUN
+    lisf1.main(prerun_file, '-v')
+
+    # DD JIRA issue https://efascom.smhi.se/jira/browse/ECC-1210 to avoid overwriting the bestrun avgdis.end.nc
+    cmd = 'cp {0}/out/avgdis{1}end.nc {0}/out/{1}/avgdis.simulated_bestend.nc'.format(subcatch.path, run_id)
```
Comment (doc78) [inline liscal/hydro_model.py:185]: Shouldn't it be cp {0}/out/{1}/avgdis.nc?

File: liscal/hydro_model.py [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+    utils.run_cmd(cmd)
+    cmd = 'cp {0}/out/lzavin{1}end.nc {0}/out/{1}/lzavin.simulated_bestend.nc'.format(subcatch.path, run_id)
```
Comment (doc78) [inline liscal/hydro_model.py:187]: same here cp {0}/out/{1}/lzavin.nc?

File: liscal/hydro_model.py [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+    utils.run_cmd(cmd)
+
+    # SECOND LISFLOOD RUN
+    lisf1.main(run_file, '-q')
+
+    # DD JIRA issue https://efascom.smhi.se/jira/browse/ECC-1210 restore the backup
+    cmd = 'rm {0}/out/avgdis{1}end.nc {0}/out/{1}/lzavin.nc'.format(subcatch.path, run_id)
```
Comment (doc78) [inline liscal/hydro_model.py:194]: and here rm {0}/out/{1}/avgdis.nc?

File: liscal/hydro_model.py [hunk 6]
```diff
@@ -0,0 +1,271 @@
...
+
+    print(">> Running LISFLOOD using the \"best\" parameter set")
+    parameters = read_parameters(subcatch.path)
+
+    run_id = 'X'
+    out_dir = os.path.join(subcatch.path_out, run_id)
+    os.makedirs(out_dir, exist_ok=True)
+    write_states = 'yes'
+
+    prerun_start = cfg.prerun_start.strftime('%d/%m/%Y %H:%M')
```
[the remark this answers is not in the document]
Comment (corentincarton) [reply liscal/hydro_model.py:175]: @doc78 you're right. I'm wondering if we shouldn't simply use forcing_start everywhere in that function... as we do the same in EFAS. prerun_start and prerun_end don't really make sense here.

[the remark this answers is not in the document]
Comment (doc78) [reply liscal/hydro_model.py]: As you prefer. In GloFAS we always use all the forcings for the longrun. I do not know in EFAS.

**Pull request merged**